Source/Mock.cs

```diff
@@ -148,18 +148,56 @@ public static void VerifyAll(params Mock[] mocks)
 		public virtual bool CallBase { get; set; }
 
 		/// <include file='Mock.xdoc' path='docs/doc[@for="Mock.DefaultValue"]/*'/>
+		[Obsolete("Replaced with DefaultValueProvider.")]
 		public virtual DefaultValue DefaultValue
 		{
 			get { return this.defaultValue; }
 			set { this.SetDefaultValue(value); }
 		}
 
+		/// <summary>
+		/// Specifies the class that will determine the default 
+		/// value to return when invocations are made that 
+		/// have no setups and need to return a default 
+		/// value (for loose mocks).
+		/// </summary>
+		public IDefaultValueProvider DefaultValueProvider
+		{
+			get { return this.defaultValueProvider; }
+			set { this.SetDefaultValueProvider(value); }
+		}
+
+		private void SetDefaultValueProvider(IDefaultValueProvider value)
+		{
+			Guard.NotNull(() => value, value);
+
+			if (value.GetType() == typeof(EmptyDefaultValueProvider))
+			{
+				this.defaultValue = DefaultValue.Empty;
+			}
+			else if (value.GetType() == typeof(MockDefaultValueProvider))
+			{
+				this.defaultValue = DefaultValue.Mock;
+			}
+			else
+			{
+				this.defaultValue = DefaultValue.Custom;
+			}
```

Comment:
Instead of faking polymorphism via a chain of `if`-based type checks, why not let a `IDefaultValueProvider` declare its kind via a `Kind` property (which is of type `DefaultValue`) so we can simply copy it over here?